issuing_using_did.md

```diff
@@ -0,0 +1,400 @@
+
+## Overview
+
+![Overview](./assets/issuing_using_did/overview.png)
+
+The proposed solution is to allow DID's to be used a terminal identifier in OA viewers.
+
+![Extending terminal identifiers](./assets/issuing_using_did/terminal_identifiers.png)
+
+The approach will extend the current approach of using only DNS as the terminal identifier and also allow for DID's. While certain identifiers such as ethereum addresses may not be human-friendly, they are still useful for machines which are able to process them. This allows anyone with DID to be able to associate it with their issued documents.
+
+Optionally, one may also associate their DID with a domain name to allow both the DID and the domain name to be recognized as the issuer.
+
+## Technical Details
+
+For simplicity sake, we will use a DID document from the universal resolver as shown in the implementation details discussed:
+
+```json
+{
+  "@context": "https://w3id.org/did/v1",
+  "id": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6",
+  "authentication": [
+    {
+      "type": "Secp256k1SignatureAuthentication2018",
+      "publicKey": ["did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6#owner"]
+    }
+  ],
+  "publicKey": [
+    {
+      "id": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6#owner",
+      "type": "Secp256k1VerificationKey2018",
+      "ethereumAddress": "0xe6fe788d8ca214a080b0f6ac7f48480b2aefa9a6",
+      "owner": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6"
+    },
+    {
+      "id": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6#delegate-1",
+      "type": "Secp256k1VerificationKey2018",
+      "owner": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6",
+      "publicKeyHex": "0295dda1dca7f80e308ef60155ddeac00e46b797fd40ef407f422e88d2467a27eb"
+    }
+  ]
+}
+```
+
+### For documents that are signed directly
+
+Assuming that an issuer with DID has the private key to one of the public key in the DID document, we can simply signal the intent to issue a given document by signing the `merkleRoot` of the document itself.
+
+The issuer will simple append this proof into the root level of the document as such. The proof structure resembles the existing proof block in [w3c vc's structure](https://www.w3.org/TR/vc-data-model/#proofs-signatures).
+
+Also, to signal if the document may or may not be revoked, a compulsory key `revocation.type` must always be present. It may take on multiple types of values such as:
+
+- `NONE`
+- `REVOCATION_STORE`
+- `SIMPLE_REVOCATION_LIST`
+- `REVOCATION_ENDPOINT`
+- etc
+
+A `revocation.type: "NONE"` must exist to ensure that the an end user did not obfuscate the revocation information to prevent his document from being revoked.
+
+```json
+{
+  "schema": "tradetrust/v1.0",
+  "data": {
+    "issuers": [
+      {
+        "name": "TradeTrust Demo",
+        "did": {
+          "id": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6",
+          "purpose": "DOCUMENT_ISSUANCE",
+          "revocation": {
```

Comment:
to move up ?

Reply:
I think it is important to be consistent with existing design as well. Existing design has the document store in the identity itself. I think it is worth considering that if we are moving to single issuer in the future we can move both the documentStore and revocationStore to the root level.